src/Xamarin.Android.Build.Tasks/Tasks/ResolveAndroidTooling.cs

```diff
@@ -285,11 +286,24 @@ bool ValidateApiLevels ()
 					// overwrite using user version only if it is 
 					// above the maxStableApi and a valid apiLevel.
 					if (userSelected != null && userSelected > maxSupported && userSelected <= maxInstalled) {
-						AndroidApiLevel = userSelected.ToString ();
-						SupportedApiLevel = userSelected.ToString ();
+						maxInstalled =
+							maxSupported = userSelected.Value;
+						AndroidApiLevel =
+							SupportedApiLevel = userSelected.ToString ();
+					}
+				}
+
+				for (int i = maxInstalled; i >= MonoAndroidHelper.SupportedVersions.MinStableVersion.ApiLevel; i--) {
```

Comment:
Hmm, I am uncertain here if this should instead be `for (int i =maxSupported;`.

If so, probably need one more test case... Maybe should switch to a parameterized test.